domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/ShapePoint.java

```diff
@@ -0,0 +1,240 @@
+ * {@link ShapePoint.ShapeBuilder} to create this.
+ */
+public class ShapePoint extends GtfsEntity implements Comparable<ShapePoint> {
+    @NotNull
+    private final String shapeId;
+    @NotNull
+    private final Float shapePtLat;
+    @NotNull
+    private final Float shapePtLon;
+    @NotNull
+    private final Integer shapePtSequence;
+    @Nullable
+    private final Float shapeDistTraveled;
+
+    /**
+     * Class for all entities defined in shapes.txt
+     *
+     * @param shapeId           identifies a shape
+     * @param shapePtLat        latitude of a shape point
+     * @param shapePtLon        longitude of a shape point
+     * @param shapePtSequence   sequence in which the shape points connect to form the shape
+     * @param shapeDistTraveled actual distance traveled along the shape from the first shape point to the point
+     *                          specified in this record
+     */
+    private ShapePoint(@NotNull final String shapeId,
+                       @NotNull final Float shapePtLat,
+                       @NotNull final  Float shapePtLon,
+                       @NotNull final Integer shapePtSequence,
+                       @Nullable final Float shapeDistTraveled) {
+        this.shapeId = shapeId;
+        this.shapePtLat = shapePtLat;
+        this.shapePtLon = shapePtLon;
+        this.shapePtSequence = shapePtSequence;
+        this.shapeDistTraveled = shapeDistTraveled;
+    }
+
+    @NotNull
+    public String getShapeId() {
+        return shapeId;
+    }
+
+    public float getShapePtLat() {
+        return shapePtLat;
+    }
+
+    public float getShapePtLon() {
+        return shapePtLon;
+    }
+
+    @NotNull
+    public Integer getShapePtSequence() {
+        return shapePtSequence;
+    }
+
+    @Nullable
+    public Float getShapeDistTraveled() {
+        return shapeDistTraveled;
+    }
+
+    /**
+     * Implement compareTo method from {@code Comparable}. Return the value {@code 0} if this {@code Shape} field
+     * shape_pt_sequence is equal to field shape_pt_sequence of argument; a value less than {@code 0} if this field
+     * shape_pt_sequence is numerically less than field shape_pt_sequence of argument; and a value greater
+     * than {@code 0} if this field shape_pt_sequence is numerically greater than field shape_pt_sequence of argument
+     *
+     * @param shapePoint shape to compare to
+     * @return the value {@code 0} if this {@code Shape} field shape_pt_sequence is equal to field shape_pt_sequence of
+     * argument; a value less than {@code 0} if this field shape_pt_sequence is numerically less than field
+     * shape_pt_sequence of argument; and a value greater than {@code 0} if this field shape_pt_sequence is numerically
+     * greater than field shape_pt_sequence of argument
+     */
+    @Override
+    public int compareTo(@NotNull final ShapePoint shapePoint) {
+        return getShapePtSequence().compareTo(shapePoint.getShapePtSequence());
```

Comment:
It seems we are comparing Integers here, so it seems calling compareTo as opposed to a plain comparison makes the code harder to read for not much of a benefit. Thoughts?

Reply:
I'd be ok leaving it as-is and depending on the `Integer.compareTo()` implementation - otherwise we're just duplicating the same if/than/else clause that's already implemented.